### PR TITLE
Add `query_account` to `jetstream::Context`

### PR DESCRIPTION
**Pull request by caspervonb**
See https://github.com/nats-io/nats-architecture-and-design/issues/120

### CODE REVIEW

File: async-nats/src/jetstream/account.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Default, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+pub struct Limits {
+    /// Maximum memory for this account (-1 if no limit)
+    pub max_memory: i64,
```
Comment (Jarema) [inline async-nats/src/jetstream/account.rs:7]: Wouldn't it be more Rusty if we would map `-1` to `None`? 

Same for all those values.

Comment (caspervonb) [reply async-nats/src/jetstream/account.rs:7]: > Wouldn't it be more Rusty if we would map -1 to None?

It would.

Comment (Jarema) [reply async-nats/src/jetstream/account.rs:7]: Is that an agreement to change it to that 😄 ?

Comment (caspervonb) [reply async-nats/src/jetstream/account.rs:7]: Yeah I'll fix it up today along with documenting the rest of the fields, these were carried over from the old definition.

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**